graylog2-web-interface/src/stores/inputs/InputStatesStore.js

```diff
@@ -28,13 +28,109 @@ const InputStatesStore = Reflux.createStore({
             }
             result[input.id][node] = input;
           });
-        })
+        });
         this.inputStates = result;
         this.trigger({inputStates: this.inputStates});
 
         return result;
       });
-  }
+  },
+
+  start(input) {
+    let url;
+
+    if (input.global) {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.ClusterInputStatesController.start(input.id).url);
+    } else {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.InputStatesController.start(input.id).url);
+    }
+
+    const handleSuccess = () => UserNotification.success(`Input '${input.title}' started successfully`);
+    const handleError = (message) => UserNotification.error(message, `Input '${input.title}' could not be started`);
+
+    return fetch('PUT', url)
+      .then(
+        () => {
+          this.list().then(inputStatesList => this._checkStartedState(input, inputStatesList, handleSuccess, handleError));
+        },
+        error => {
+          handleError(`Error starting input '${input.title}': ${error}`);
+        });
+  },
+
+  stop(input) {
+    let url;
+
+    if (input.global) {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.ClusterInputStatesController.stop(input.id).url);
+    } else {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.InputStatesController.stop(input.id).url);
```

Comment:
What happens when the input is not global, but not running on the node we're connected too?

I think we should not differentiate the input type (global/non-global) on the frontend side, but in the ClusterInputStatesController, which has far more knowledge to determine the termination strategy.